packages/payment-detection/src/near-detector.ts

```diff
@@ -58,35 +63,55 @@ export default class NearNativeTokenPaymentDetector extends ReferenceBasedDetect
   };
 
   /**
-   * Extracts the events for an address and a payment reference
+   * Extracts payment events of an address matching an address and a payment reference
    *
-   * @private
-   * @param address Payment address
-   * @param eventName Is it for payment or refund
-   * @param network The payment network name
+   * @param address Address to check
+   * @param eventName Indicate if it is an address for payment or refund
+   * @param requestCurrency The request currency
    * @param paymentReference The reference to identify the payment
-   * @param paymentNetworkVersion
-   * @returns The balance with events
+   * @param paymentNetwork the payment network
+   * @returns The balance
    */
   protected async extractEvents(
     address: string,
     eventName: PaymentTypes.EVENTS_NAMES,
-    network: string,
+    requestCurrency: RequestLogicTypes.ICurrency,
```

Comment:
Same comment here, `NearDetector` is used for `NativeToken` requests so I don't understand why we would precise the request currency.